core/src/main/scala/akka/persistence/r2dbc/internal/codec/TimestampCodec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal.codec
+
+import java.time.Instant
+import java.time.LocalDateTime
+import java.util.TimeZone
+
+import io.r2dbc.spi.Row
+import io.r2dbc.spi.Statement
+
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.internal.InstantFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] sealed trait TimestampCodec {
+
+  def encode(timestamp: Instant): Any
+  def decode(row: Row, name: String): Instant
+  def decode(row: Row, index: Int): Instant
+
+  // should we name it just `now()`? The type should not be in the name...
+  def instantNow(): Instant = InstantFactory.now()
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object TimestampCodec {
+
+  class PostgresTimestampCodec extends TimestampCodec {
+    override def decode(row: Row, name: String): Instant = row.get(name, classOf[Instant])
+    override def decode(row: Row, index: Int): Instant = row.get(index, classOf[Instant])
+
+    override def encode(timestamp: Instant): Any = timestamp
+  }
+  object PostgresTimestampCodec extends PostgresTimestampCodec
+
+  case object SqlServerTimestampCodec extends TimestampCodec {
+
+    // should this come from config?
+    private val zone = TimeZone.getTimeZone("UTC").toZoneId
+
+    private def toInstant(timestamp: LocalDateTime) =
+      timestamp.atZone(zone).toInstant
+
+    override def decode(row: Row, name: String): Instant = toInstant(row.get(name, classOf[LocalDateTime]))
+
+    override def encode(timestamp: Instant): LocalDateTime = LocalDateTime.ofInstant(timestamp, zone)
```

Comment:
this is good, it wouldn't surprise me if r2dbc-sqlserver would add support for binding Instant in a future version. seems like a missing thing to me. anyway, for us this is good now

Reply:
They discussed it already [here](https://github.com/r2dbc/r2dbc-mssql/issues/174) and [here](https://github.com/r2dbc/r2dbc-mssql/issues/278).